apps/cowswap-frontend/src/modules/trade/utils/swapErrorHelper.ts

```diff
@@ -1,11 +1,19 @@
 import { getProviderErrorMessage, isRejectRequestProviderError } from '@cowprotocol/common-utils'
 
+import { getIsOrderBookTypedError } from 'api/gnosisProtocol'
+
 export const USER_SWAP_REJECTED_ERROR = 'User rejected signing the order'
 
 export function getSwapErrorMessage(error: Error): string {
   if (isRejectRequestProviderError(error)) {
     return USER_SWAP_REJECTED_ERROR
   } else {
-    return getProviderErrorMessage(error)
+    const defaultErrorMessage = getProviderErrorMessage(error)
+
+    if (getIsOrderBookTypedError(error)) {
+      return error.body?.description || defaultErrorMessage
```

Comment:
Can you capitalize the first word in case of the default API response?

Reply:
Makes sense, added